pkg/cmd/stack/deploy.go

```diff
@@ -586,8 +589,7 @@ func deployK8sService(ctx context.Context, svcName string, s *model.Stack, c kub
 	}
 
 	svcK8s.ObjectMeta.ResourceVersion = old.ObjectMeta.ResourceVersion
-	driver := divert.New(nil, nil, nil)
-	driver.ApplyToService(svcK8s, old)
+	keepDivertInService(svcK8s, old)
```

Comment:
I moved this logic to the weaver driver in [this PR](https://github.com/okteto/okteto/pull/3288) but I moved it back to the stack module because it was complicating the divert driver interface. This is an optimization we do for stacks, and we will probably remove the weaver driver soon, so I think it's better to move it back here and keep the divert driver interface simpler

Reply:
yes, this is weaver specific. but this logic will be removed soon and it was complicating the divert driver interface